python/ray/util/sgd/v2/worker_group.py

```diff
@@ -37,6 +60,20 @@ def __init__(self, *args, **kwargs):
         return _WrappedExecutable
 
 
+def construct_metadata() -> WorkerMetadata:
+    """Creates metadata for this worker.
+
+    This function is expected to be run on the actor.
+    """
+    node_id = ray.get_runtime_context().node_id.hex()
+    node_ip = ray.util.get_node_ip_address()
```

Comment:
Do we need both `node_id` and `node_ip`? More generally, do we have best practices around when to use one vs. the other since I presume these should be 1:1?

Reply:
There's no harm in having both. Yes they should be 1:1.